google_guest_agent/metadata.go

```diff
@@ -230,5 +266,8 @@ func getMetadata(ctx context.Context, hang bool) (*metadata, error) {
 		return nil, err
 	}
 	var ret metadata
-	return &ret, json.Unmarshal(md, &ret)
+	if err := json.Unmarshal(md, &ret); err != nil {
+		return nil, err
```

Comment:
Why not:
```
err := json.Unmarshal(md, &ret)
return &ret, err
```

Reply:
reverted as we dont do the jsonerr herr anymore